clients/application/src/main/java/org/eclipse/hono/application/client/AsyncCommandClient.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.application.client;
+
+import java.util.Map;
+
+import org.eclipse.hono.client.ServiceInvocationException;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * A client for sending asynchronous commands.
+ *
+ * @see <a href="https://www.eclipse.org/hono/docs/api/command-and-control/">
+ *      Command &amp; Control API for AMQP 1.0 Specification</a>
+ * @see <a href="https://www.eclipse.org/hono/docs/api/command-and-control-kafka/">
+ *      Command &amp; Control API for Kafka Specification</a>
+ */
+public interface AsyncCommandClient {
+    /**
+     * Sends an async command to a device, i.e. there is no immediate response expected from the device, but
+     * asynchronously via a separate consumer.
+     * <p>
+     * A device needs to be (successfully) registered before a client can upload any data for it. The device also needs
+     * to be connected for a successful delivery.
+     *
+     * @param deviceId The device to send the command to.
+     * @param command The command name.
+     * @param data The command data to send to the device or {@code null} if the command has no input data.
+     * @param correlationId The identifier to use for correlating the response with the request. Note: This parameter is
+     *            security sensitive. To ensure secure request response mapping choose correlationId carefully, e.g.
+     *            {@link java.util.UUID#randomUUID()}.
+     * @param replyId This identifier is to be set depending upon the underlying implementation (i.e AMQP or Kafka based).
+     *                <ul>
+     *                <li>In Kafka based implementation, this is not used and hence it can be set as {@code null}.</li>
+     *                <li>In AMQP based implementation, this is an arbitrary string which gets used for the response 
+     *                link address in the form of <em>command_response/${tenantId}/${replyId}</em>. Must match the
```

Comment:
Is there any support method for creating a consumer for that address anywhere?

Reply:
My intention is to keep this PR small so that it will be easier to review. Hence I added only the `AsyncCommandClient` part in this PR. The consumer part is planned for the next PR.